src/openapi/streaming/subscription.js

```diff
@@ -218,13 +228,20 @@ function onSubscribeError(referenceId, response) {
         return;
     }
 
-    this.currentState = STATE_UNSUBSCRIBED;
-    log.error(LOG_AREA, 'An error occurred subscribing', { response, url: this.url });
+    if (response && response.status === '504' && this.retryCounter++ < MAX_RETRIES) {
+        log.error(LOG_AREA, 'Received gateway timeout when subscribing. Retrying.', { response, url: this.url });
 
-    // if we are unsubscribed, do not fire the error handler
-    if (this.queue.peek() !== ACTION_UNSUBSCRIBE) {
-        if (this.onError) {
-            this.onError(response);
+        this.currentState = STATE_SUBSCRIBED; // set to subscribed so that unsubscribe request isn't ignored
+        retrySubscription.call(this);

```

Comment:
If we indeed need to call unsubscribe first to subscribe after subscribe timeout. We should first peek into next action. If next action is unsubscribe, no need to try to retry subscription. 
```
this.queue.peek()
```